### PR TITLE
Add a man page written in asciidoc

### PR DESCRIPTION
**Pull request by musicinmybrain**
Automatically generate and install it when using meson

### CODE REVIEW

**Comment by musicinmybrain** [comment]
See https://github.com/grimme-lab/crest/issues/56.

I’m happy to fine-tune this PR based on any feedback.

File: docs/man/meson.build [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        asciidoctor,
+	'--attribute=release-version=@0@'.format(meson.project_version()),
+	'--backend=manpage',
+	'--doctype=manpage',
+	'--out-file=@OUTPUT@',
+	'@INPUT@',
+      ],
+    )
+  endforeach
+endif
```
Comment (awvwgk) [inline docs/man/meson.build:37]:
```suggestion
asciidoctor = find_program('asciidoctor', required: false)
if asciidoctor.found()
  foreach man_name : ['crest.adoc']
    install_man(
      configure_file(
        input: files(man_name),
        output: '@BASENAME@.1',
        command: [
          asciidoctor,
          '--attribute=release-version=@0@'.format(meson.project_version()),
          '--backend=manpage',
          '--out-file=@OUTPUT@',
          '@INPUT@',
        ],
      )
    )
  endforeach
endif
```

File: meson_options.txt [hunk 2]
```diff
@@ -27,3 +27,9 @@ option(
   value: [],
   description: 'libraries to load for custom linear algebra backend',
 )
+option(
+  'man_pages',
+  type: 'boolean',
+  value: 'true',
+  description: 'whether to generate man page(s) (requires asciidoctor)',
+)
```
Comment (awvwgk) [inline meson_options.txt:35]:
```suggestion
```

**Review by awvwgk — COMMENTED** [review]
First, thanks for sharing, the man page looks great.

I suggest to use `configure_file` instead of a custom target, since `configure_file` will rerun automatically for changes in the asciidoc source file, while a custom target requires explicit action. Also, the man page should be installed with the `install_man` command to make sure it is included with the distribution.

Finally, I suggest to remove the option and just check for the availability of the `asciidoctor` processor, in case it is available we build the man page, otherwise we skip it (similar behavior is used in `xtb`).

Regarding CMake integration I fully agree. Converting man pages and installing them properly is easy in meson, but requires an incredible amount of boilerplate in CMake and I don't think there is much value in doing this exercise in CMake.

**Review by pprcht — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**